core-rust/state-manager/src/protocol/protocol_update.rs

```diff
@@ -0,0 +1,489 @@
+                last_batch_idx: *batch_idx,
+            },
+        }
+    }
+
+    pub fn next_committable_batch_idx(&self) -> Option<u32> {
+        match self.read_protocol_update_progress() {
+            ProtocolUpdateProgress::UpdateInitiatedButNothingCommitted {
+                protocol_version_name: state_protocol_version_name,
+            } => {
+                if self.protocol_version_name == state_protocol_version_name {
+                    Some(0)
+                } else {
+                    None
+                }
+            }
+            ProtocolUpdateProgress::UpdateInProgress {
+                protocol_version_name: state_protocol_version_name,
+                last_batch_idx,
+            } => {
+                if self.protocol_version_name == state_protocol_version_name {
+                    Some(last_batch_idx.checked_add(1).unwrap())
+                } else {
+                    None
+                }
+            }
+            ProtocolUpdateProgress::NotUpdating => None,
+        }
+    }
+
+    pub fn commit_flash(&mut self, state_updates: StateUpdates) {
+        let nonce = self.store.read_current().max_state_version().number();
+        let flash_txn = LedgerTransaction::FlashV1(Box::new(FlashTransactionV1 {
+            nonce,
+            state_updates,
+        }));
+        self.commit_txn_batch(vec![flash_txn]);
+    }
+
+    fn commit_txn_batch(&mut self, transactions: Vec<LedgerTransaction>) {
+        let batch_idx = self
+            .next_committable_batch_idx()
+            .expect("Can't commit next protocol update batch");
+
+        let read_store = self.store.read_current();
+        let latest_proof: LedgerProof = read_store
+            .get_latest_proof()
+            .expect("Pre-genesis protocol updates are currently not supported");
+
+        let lock_factory = LockFactory::new("protocol_update");
+        let execution_cache = lock_factory.new_mutex(ExecutionCache::new(
+            latest_proof.ledger_header.hashes.transaction_root,
+        ));
+        // For the purpose of executing protocol update transactions we're just going to use
+        // a dummy protocol state with no configured updates and the name of this (in progress)
+        // protocol update as the current version (although that could really be any string,
+        // it doesn't matter here).
+        let dummy_protocol_state = ProtocolState {
+            current_epoch: None,
+            current_protocol_version: self.protocol_version_name.clone(),
+            enacted_protocol_updates: btreemap!(),
+            unenacted_protocol_updates: vec![],
+        };
+
+        let mut series_executor = TransactionSeriesExecutor::new(
+            read_store.deref(),
+            &execution_cache,
+            &self.execution_configurator,
+            dummy_protocol_state,
+        );
+
+        let mut committed_transaction_bundles = Vec::new();
```

Comment:
The ~70 lines starting here mimic the `StateComputer.commit()` (quite understandably :p).
We definitely were incrementally adding stuff to that logic, and it may happen that we add more. Forgetting about protocol update's "copy" would be very sad (and possibly only caught at next protocol update).

Hence I think it is a good moment to actually address the TODO left there: https://github.com/radixdlt/babylon-node/blob/8445409a552b176a0d7c23b220208b25603f243c/core-rust/state-manager/src/state_computer.rs#L1135 (and then re-use the resultant "committer" helper here)

Since we are in a rush, we can live with the copy-paste + another `TODO(urgent)` here, for a moment.